rootfs/etc/nginx/lua/test/balancer/sticky_test.lua

```diff
@@ -40,7 +40,7 @@ end
 
 describe("Sticky", function()
   before_each(function()
-    mock_ngx({ var = { location_path = "/", host = "test.com" } })
+    mock_ngx({ var = { location_path = "/", host = "test.com", https = "on" } })
```

Comment:
instead of changing this globally please introduce another test just like https://github.com/kubernetes/ingress-nginx/pull/3509/files#diff-f62b36fb1d17870f3fbc347d5624fc1fR112 but for when ngx.var.https is "on".

Then inside that specific test case (`it`) you can mock `$https` `using ngx.var.https = "on"`